src/templates/keyTransform.ts

```
@@ -0,0 +1,173 @@
+        if (!key || typeof key !== 'string') {
+            return false;
+        }
+        
+        // Check length (Kafka recommends keys under 1KB, but shorter is better)
+        if (key.length > 255) {
+            console.warn(`Key is quite long (${key.length} chars): ${key.substring(0, 50)}...`);
+            return false;
+        }
+        
+        // Check for problematic characters that might cause issues
+        const problematicChars = /[<>:"/\\|?*\x00-\x1f]/;
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)